src/config/util/tls.rs

```diff
@@ -0,0 +1,71 @@
+use anyhow::{Context, Error, Result};
+use rustls::internal::pemfile;
+use rustls::{Certificate, PrivateKey};
+use serde::Deserialize;
+use std::fs::File;
+use std::io::BufReader;
+use std::path::Path;
+use std::str::FromStr;
+
+#[derive(Clone, Debug, Deserialize, PartialEq)]
+pub enum PrivateKeyAlgorithm {
+    #[serde(rename = "rsa")]
+    Rsa,
+    #[serde(rename = "pkcs8")]
+    Pkcs8,
+}
+
+impl FromStr for PrivateKeyAlgorithm {
+    type Err = Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "rsa" => Ok(PrivateKeyAlgorithm::Rsa),
+            "pkcs8" => Ok(PrivateKeyAlgorithm::Pkcs8),
+            _ => Err(Error::msg(format!(
+                "Invalid algorithm name provided for TLS key. {}",
+                s
+            ))),
+        }
+    }
+}
+
+pub fn load_cert(path: &Path) -> Result<Vec<Certificate>> {
+    let file = File::open(path.to_path_buf()).context(format!(
+        "Unable to find the TLS certificate on: {}",
+        path.to_str().unwrap()
+    ))?;
+    let mut reader = BufReader::new(file);
+
+    pemfile::certs(&mut reader).map_err(|_| {
+        let path = path.to_str().unwrap();
+
+        Error::msg(format!("Failed to read certificates at {}", path))
```

Comment:
Similarly, I think this can be improved with [`anyhow!`](https://docs.rs/anyhow/1/anyhow/macro.anyhow.html):

```suggestion
        anyhow!("Failed to read certificates at {}", path)
```

This applies to other usages of `Error::msg(format!(...))` too.